src/server.cpp

```diff
@@ -593,6 +595,32 @@ void Server::AsyncRunStep(bool initial_step)
 	}
 
 	/*
+		Listen to the admin chat, if available
+	*/
+	if (m_admin_chat) {
+		while (!m_admin_chat->command_queue.empty()) {
+			ChatEvent evt = m_admin_chat->command_queue.pop_frontNoEx();
+			if (evt.type == CET_NICK_ADD) {
+				// The terminal informed us of its nick choice
+				m_admin_nick = evt.nick;
+				// Check that an account exists, and inform that an account has to be created
+				// TODO: remove this if its decided to go with a more "tolerant" auth handler
+				/*if (!m_script->getAuth(m_admin_nick, NULL, NULL)) {
+					errorstream << "You haven't set up an account." << std::endl
+						<< "Please log in using the client as '"
+						<< m_admin_nick << "' with a secure password." << std::endl
+						<< "Until then, you can't execute admin tasks via the console either." << std::endl;
+				}//*/
+			} else {
+				assert(evt.type == CET_CHAT);
```

Comment:
Is this PR still a work-in-progress??
I can't help but wonder if this small piece of code indicates there are more chat event types to be supported.